clippy_lints/src/operators/mod.rs

```diff
@@ -837,6 +838,49 @@ declare_clippy_lint! {
     "explicit self-assignment"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    ///
+    /// Checks for assignments via raw pointers that involve types with destructors.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Assignments of the form `*ptr = new_value;` assume that `*ptr` contains an initialized
+    /// value, and unconditionally execute the `std::ops::Drop`-implementation if such
+    /// implementation is defined on the type of `*ptr`. If the value is in fact
+    /// uninitialized or otherwise invalid, the execution of `std::ops::Drop::drop(&mut self)`
+    /// is always Undefined Behavior.
+    ///
+    /// Use `std::ptr::write()` to overwrite a value without executing the destructor.
+    ///
+    /// Use `std::ptr::drop_in_place()` to conditionally execute the destructor if you are
+    /// sure that the place contains an initialized value.
+    ///
+    /// ### Example
+    /// ```no_run
+    /// unsafe fn foo(oldvalue: *mut String) {
+    ///     // Direct assignment always executes `String`'s destructor on `oldvalue`
+    ///     *oldvalue = "New Value".to_owned();
```

Comment:
You should use an `unsafe` block inside the function, even if this is an `unsafe` function, as this example will warn in edition 2024:

```
warning[E0133]: dereference of raw pointer is unsafe and requires unsafe block
 --> t.rs:4:5
  |
4 |     *oldvalue = "New Value".to_owned();
  |     ^^^^^^^^^ dereference of raw pointer
  |
  = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/unsafe-op-in-unsafe-fn.html>
  = note: raw pointers may be null, dangling or unaligned; they can violate aliasing rules and cause data races: all of these are undefined behavior
note: an unsafe function restricts its caller, but its body is safe by default
```

Reply:
Thanks for the heads-up